MIGRATION_GUIDE.md

```diff
@@ -1,14 +1,264 @@
-# [WIP] N-API Migration Guide
+# N-API Migration Guide
```

Comment:
Should we name this guide as `1.0 Migration`? There seems to be some things that aren't related to NAPI (from my understanding) like event queue.

Reply:
They're all necessary for migrating to the N-API backend, but it's a good idea to be clearer that these changes will be permanent for 1.0.

Reply:
Maybe we can leave that note and then rename it the "1.0 Migration Guide" as part of that release?

Reply:
Let's revisit that as we get close to 1.0 launch.